flatnav/Index.h

```diff
@@ -59,27 +58,26 @@ template <typename dist_t, typename label_t> class Index {
 
   ~Index() { delete[] _index_memory; }
 
-  bool add(void *data, label_t &label, int ef_construction,
+  void add(void *data, label_t &label, int ef_construction,
```

Comment:
I think it's better engineering-wise to raise exceptions when things go wrong rather than return a boolean variable since we're able to capture errors early!